internal/client/client.go

```diff
@@ -57,6 +58,34 @@ type Config struct {
 	StatsPeriod   time.Duration
 }
 
+func NewConfig() *Config {
```

Comment:
Side note: You did not list any config related changes in this PR's description. Are these config related changes intentional? Also can you explain why they are needed.

Reply:
There are no config related changes. I've mentioned about the internal client refactoring in the description and this config is only for the internal client. There's no public API change for the config.

Reply:
Clarification: The internal client was added at: https://github.com/hazelcast/hazelcast-go-client/pull/695